cassiopeia/tools/branch_length_estimator/_iid_exponential_mle.py

```diff
@@ -292,8 +365,17 @@ def estimate_branch_lengths(self, tree: CassiopeiaTree) -> None:
         prob = cp.Problem(obj, all_constraints)
         try:
             prob.solve(solver=solver, verbose=verbose)
-        except cp.SolverError:  # pragma: no cover
-            raise IIDExponentialMLEError("Third-party solver failed")
+            self._backup_solver_was_needed = False
+        except cp.SolverError as err:  # pragma: no cover
+            # We try the backup_solver
+            try:
+                if backup_solver is None:
+                    # We don't retry; just raise the original error
+                    raise err
+                prob.solve(solver=backup_solver, verbose=verbose)
+                self._backup_solver_was_needed = True
+            except cp.SolverError:  # pragma: no cover
+                raise IIDExponentialMLEError("Third-party solver failed")
```

Comment:
Maybe specifically mention that the backup failed? Or if you think this is sufficiently covered by this coupled with _backup_solver_was_needed then it's fine.